samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -298,6 +298,7 @@ public void releaseResources(SamzaResource resource) {
       }
       amClient.releaseAssignedContainer(container.getId());
       allocatedResources.remove(resource);
+      metrics.decrementAllocatedContainersInBuffer();
```

Comment:
Do we want to decrement this metric in the stopStreamProcessor method as well?

Reply:
I want this metric to always in sync with "allocatedResources" in this class.
stopStreamProcessor doesn't remove container from allocatedResources.
Actually, in stopStreamProcessor. it calls amClient.releaseAssignedContainer(yarnContainer.id()) only when allocatedResources.get(resource) == null which means the container was not in the "allocatedResources" Map already